src/items/rqgItem.ts

```diff
@@ -182,13 +187,28 @@ export class RqgItem extends Item {
 
     const compendiumItem = await this.getItemFromAllCompendiaByRqid(rqid, lang);
 
-    return compendiumItem;
+    if (compendiumItem !== undefined) {
+      return compendiumItem;
+    } else {
+      const msg = localize("RQG.Item.RqgItem.Error.ItemNotFoundByRqid", {
+        rqid: rqid,
+        rqidLang: lang,
+      });
+      ui.notifications?.warn(msg);
+      console.log(msg);
+    }
+
   }
 
   private static async getItemFromWorldByRqid(
     rqid: string,
     lang: string = "en"
   ): Promise<RqgItem | undefined> {
+
+    if (!rqid || rqid === "") {
```

Comment:
The second part is unnecessary since empty string is "falsy"
```suggestion
    if (!rqid) {
```